### DISCUSSION TITLE
Is there a way to discover the current key bindings for a command?

### DISCUSSION THREAD

**Discussion by skybrian** [microsoft/vscode-discussions#780] [Extension Development QnA] [answered]
I have an extension with some placeholder text, added using a text editor decoration:

> Type your question here and press Control+Alt Enter to get a response

However, the user might have changed the key binding. Is there some way to find out what the actual key binding is for a command? Ideally the message would be customized for the actual keyboard they're using.

I don't see any API related to key bindings. (I asked GPT4 and it hallucinated one.)

**Comment by theReynald** [comment, marked as answer]
Hi @skybrian

I'm not aware that there is a direct API in VS Code for querying the current keybindings from within an extension.

However, you can consider the following workaround to accomplish this:

1. **Load Keybindings from User Settings**:

   - VS Code stores its keybindings in a JSON file. The extension can read the user's keybindings by loading the `keybindings.json` file from the user's settings directory. The path to this file can vary based on the platform (Windows, Mac, or Linux).

   - Once you've read the file, you can search within the JSON data for the command you're interested in and then extract the keybinding.

2. **Keep a Default Text**: 

   If you're unable to determine the custom keybinding or if the user has not set a custom keybinding for the command, then you can display a default text.

Hope this help a bit.

**Reply by kschaab** [reply]
This only works when the user is not using remote like remote ssh.  If the user is using a remote then the keybindings are coming from the client OS but extensions are executing on the remote OS.  This will result in the extension getting the improper keybindings.  It would be great if VS Code could export keybindings through an API even if it is just to return the user keybindings and default keybindings json separately it would be extremely helpful.
